parser/grammar.y

```diff
@@ -55,6 +58,94 @@ type stringModifiers struct {
   XorMax int32
 }
 
+// Lexer is an adapter that fits the flexgo lexer ("Scanner") into goyacc
+type lexer struct {
+	scanner Scanner
+	err     gyperror.Error
+	ruleSet *ast.RuleSet
+}
+
+// Lex provides the interface expected by the goyacc parser.
+// It sets the context's lval pointer (defined in the lexer file)
+// to the one passed as an argument so that the parser actions
+// can make use of it.
+func (l *lexer) Lex(lval *yrSymType) int {
+	l.scanner.Context.lval = lval
+	r := l.scanner.Lex()
+	if r.Error.Code != 0 {
+		r.Error.Line = l.scanner.Lineno
+		panic(r.Error)
+	}
+	return r.Token
+}
+
+// Error satisfies the interface expected of the goyacc parser.
+func (l *lexer) Error(msg string) {
+	l.err = gyperror.Error{
+		Code:    gyperror.LexicalError,
+		Line:    l.scanner.Lineno,
+		Message: msg,
+	}
+}
+
+// SetError sets the lexer error. The error message can be built by passing
+// a format string and arguments as fmt.Sprintf. This function returns 1 as
+// it's intended to by used in grammar.y as:
```

Comment:
- I think that we can stop exporting `SetError`.
- This function returns 1 as it's intended to be used  ~~in grammar.y~~ as: